add docs for nchs-mortality (#264)
cmu-delphi/delphi-epidata: docs/api/covidcast-signals/nchs-mortality.md

| `deaths_flu_incidence_num` | Number of weekly new deaths involving Influenza |
| `deaths_flu_incidence_prop` | Number of weekly new deaths involving Influenza, per 100,000 population |
| `deaths_pneumonia_notflu_incidence_num` | Number of weekly new deaths involving Pneumonia, excluding Influenza deaths |
| `deaths_pneumonia_notflu_incidence_prop` | Number of weekly new deaths involving Pneumonia, excluding Influenza deaths, per 100,000 population |
| `deaths_covid_and_pneumonia_notflu_incidence_num`| Number of weekly new deaths involving COVID-19 and Pneumonia, excluding Influenza |
| `deaths_covid_and_pneumonia_notflu_incidence_prop`| Number of weekly new deaths involving COVID-19 and Pneumonia, excluding Influenza, per 100,000 population |
|`deaths_pneumonia_or_flu_or_covid_incidence_num`| Number of weekly new deaths involving Pneumonia, Influenza, or COVID-19|
|`deaths_pneumonia_or_flu_or_covid_incidence_prop`| Number of weekly new deaths involving Pneumonia, Influenza, or COVID-19, per 100,000 population|
|`deaths_percent_of_expected`| The number of weekly new deaths for all causes in 2020 compared to the average number across the same week in 2017–2019|

> These signals are taken directly from [Table 1](https://www.cdc.gov/nchs/nvss/vsrr/COVID19/index.htm) without changes. National provisional death counts include deaths occurring within the 50 states and the District of Columbia that have been received and coded as of the date specified during a given time period. The deaths are classified based on a new ICD-10 code. (Note that, the classification is based on all the codes on the death certificate, but not the primary cause of death). The codes that are considered for each signals are described in details [here](https://github.com/cmu-delphi/covidcast-indicators/blob/nchs_mortality/nchs_mortality/DETAILS.md#metrics-level-1-m1). We export the state-level data as-is in a weekly format. 

Reviewer: ```suggestion
These signals are taken directly from [Table 1](https://www.cdc.gov/nchs/nvss/vsrr/COVID19/index.htm) without changes. National provisional death counts include deaths occurring within the 50 states and the District of Columbia that have been received and coded as of the date specified during a given time period. The deaths are classified based on a new ICD-10 code. (Note that the classification is based on all the codes on the death certificate, not just the primary cause of death). The codes that are considered for each signals are described in detail [here](https://github.com/cmu-delphi/covidcast-indicators/blob/nchs_mortality/nchs_mortality/DETAILS.md#metrics-level-1-m1). We export the state-level data as-is in a weekly format. 
```

Reviewer 2: ```suggestion
These signals are taken directly from [Table 1](https://www.cdc.gov/nchs/nvss/vsrr/COVID19/index.htm) without changes. National provisional death counts include deaths occurring within the 50 states and the District of Columbia that have been received and coded as of the date specified during a given time period. The deaths are classified based on a new ICD-10 code. (Note that the classification is based on all the codes on the death certificate, not just the primary cause of death). The codes that are considered for each signal are described in detail [here](https://github.com/cmu-delphi/covidcast-indicators/blob/nchs_mortality/nchs_mortality/DETAILS.md#metrics-level-1-m1). We export the state-level data as-is in a weekly format.
```

## Table of contents
{: .no_toc .text-delta}

1. TOC
{:toc}

## Geographical Exceptions

New York City is considered as a special state in the NCHS Mortality data, but we don't consider NYC separately. The death counts for NYC are included in New York State in our reports.

## Report Using Epiweeks

We report the NCHS Mortality data in a weekly format (`time_type=week` \& `time_value=\{YYYYWW\}`, where `YYYYWW` refers to an epiweek). The CDC defines the [epiweek](https://wwwn.cdc.gov/nndss/document/MMWR_Week_overview.pdf) as seven days, from Sunday to Saturday. We check the week-ending dates provided in the NCHS morality data and use Python package [epiweeks](https://pypi.org/project/epiweeks/) to convert them into epiweek format.

## Lag and Backfill

> There is a lag in time between the death occurred and when the death certificate is completed, submitted to NCHS and processed for reporting purposes. The death counts for earlier weeks are continually revised and may increase or decrease as new and updated death certificate data are received from the states by NCHS.  This delay can range from 1 week to 8 weeks or even more. We check for updates every weekday as they are reported by NCHS, but will report the signals weekly (on Mondays). The changes in the data due to backfill made over the past week can be fetched from our API.

Reviewer: ```suggestion
There is a lag in time between when the death occurred and when the death certificate is completed, submitted to NCHS, and processed for reporting purposes. The death counts for earlier weeks are continually revised and may increase or decrease as new and updated death certificate data are received from the states by NCHS. This delay can range from 1 week to 8 weeks or even more. We check for updates reported by NCHS every weekday but will report the signals weekly (on Monday). The data changes due to backfill made over the past week can be fetched from our API.
```

| `deaths_flu_incidence_num` | Number of weekly new deaths involving Influenza |
| `deaths_flu_incidence_prop` | Number of weekly new deaths involving Influenza, per 100,000 population |
| `deaths_pneumonia_notflu_incidence_num` | Number of weekly new deaths involving Pneumonia, excluding Influenza deaths |
| `deaths_pneumonia_notflu_incidence_prop` | Number of weekly new deaths involving Pneumonia, excluding Influenza deaths, per 100,000 population |
| `deaths_covid_and_pneumonia_notflu_incidence_num`| Number of weekly new deaths involving COVID-19 and Pneumonia, excluding Influenza |
| `deaths_covid_and_pneumonia_notflu_incidence_prop`| Number of weekly new deaths involving COVID-19 and Pneumonia, excluding Influenza, per 100,000 population |
|`deaths_pneumonia_or_flu_or_covid_incidence_num`| Number of weekly new deaths involving Pneumonia, Influenza, or COVID-19|
|`deaths_pneumonia_or_flu_or_covid_incidence_prop`| Number of weekly new deaths involving Pneumonia, Influenza, or COVID-19, per 100,000 population|
|`deaths_percent_of_expected`| The number of weekly new deaths for all causes in 2020 compared to the average number across the same week in 2017–2019|

> These signals are taken directly from [Table 1](https://www.cdc.gov/nchs/nvss/vsrr/COVID19/index.htm) without changes. National provisional death counts include deaths occurring within the 50 states and the District of Columbia that have been received and coded as of the date specified during a given time period, especially for the more recent time periods. The deaths are classified based on a new ICD-10 code. (Note that, the classification is based on all the codes on the death certificate, but not the primary cause of death). The codes that are considered for each signals are described in details [here](https://github.com/cmu-delphi/covidcast-indicators/blob/nchs_mortality/nchs_mortality/DETAILS.md#metrics-level-1-m1). We export the state-level data as-is in a weekly format. 

Author: This is incorrectly left there. Will delete them.

---
title: NCHS Mortality Data
parent: Data Sources and Signals
grand_parent: COVIDcast API
---

# NCHS Mortality Data
{: .no_toc}

* **Source name:** ``
* **Number of data revisions since 19 May 2020:** 0
* **Date of last change:** Never
* **Available for:** state (see [geography coding docs](../covidcast_geography.md))

This data source of Provisional Death Counts for COVID-19 is based on reports made
available by National Center for Health statistics [(NCHS)](https://www.cdc.gov/nchs/nvss/vsrr/COVID19/index.htm).

| Signal | Description |
| --- | --- |

> | `covid_deaths_num` | Number of all the weekly new deaths with confirmed or presumed COVID-19 |
> | `covid_deaths_prop` | Number of all the weekly new deaths with confirmed or presumed COVID-19 per 100,000 population |

Reviewer 3: ```suggestion
| `covid_deaths_num` | Number of weekly new deaths with confirmed or presumed COVID-19 |
| `covid_deaths_prop` | Number of weekly new deaths with confirmed or presumed COVID-19, per 100,000 population |
```

| `total_deaths_num` | Number of weekly new deaths from all causes |
| `total_deaths_prop` | Number of weekly new deaths from all causes per 100,000 population |
| `influenza_deaths_num` | Number of weekly new deaths involving Influenza |
| `influenza_deaths_prop` | Number of weekly new deaths involving Influenza per 100,000 population |
| `pneumonia_deaths_num` | Number of weekly new deaths involving Pneumonia, excluding Influenza deaths |
| `pneumonia_deaths_prop` | Number of weekly new deaths involving Pneumonia, excluding Influenza deaths per 100,000 population |
| `pneumonia_and_covid_deaths_num`| Number of weekly new deaths involving COVID-19 and Pneumonia, excluding Influenza |
|`pneumonia_and_covid_deaths_prop`| Number of weekly new deaths involving COVID-19 and Pneumonia, excluding Influenza per 100,000 population |
|`pneumonia_influenza_or_covid_19_deaths_num`| Number of weekly new deaths involving Pneumonia, Influenza, or COVID-19|
|`pneumonia_influenza_or_covid_19_deaths_prop`| Number of weekly new deaths involving Pneumonia, Influenza, or COVID-19 per 100,000 population|

> |`percent_of_expected_deaths`| The number of weekly new deaths for all causes in 2020 compared to the average number across the same week in 2017–2019|

Reviewer 3: This is vague; how is it compared, and what is the percent of?

Reviewer: I believe it's literally

` 100 * (number of weekly new deaths for all causes in 2020) / (average number of weekly new deaths for all causes across the same week in 2017–2019)`

These signals are taken directly from [Table 1](https://www.cdc.gov/nchs/nvss/vsrr/COVID19/index.htm) without changes. National provisional counts include deaths occurring within the 50 states and the District of Columbia that have been received and coded as of the date specified. The deaths are classified based on a new ICD-10 code. The codes that are considered for each signals are described in details [here](https://github.com/cmu-delphi/covidcast-indicators/blob/nchs_mortality/nchs_mortality/DETAILS.md#metrics-level-1-m1). We export the state-level data as-is in a weekly format. 

## Table of contents
{: .no_toc .text-delta}

1. TOC
{:toc}

## Geographical Exceptions

> New York City is considered as a special state in the NCHS Mortality data, but we don't consider NYC separately. The death counts for NYC would be included in New York State in our reports.

Reviewer 3: ```suggestion
New York City is considered as a special state in the NCHS Mortality data, but we don't consider NYC separately. The death counts for NYC are included in New York State in our reports.
```

## Report Using Epiweeks

> We report the NCHS Mortality data in a weekly format (weekly_YYYYWW, where YYYYWW refers to an epiweek). However, NCHS reports their weekly data from Saturday to Saturday. We assume there is a one day shift. For example, they report a death counts for Alaska in a week starting from date D, we will report the timestamp of this report as the corresponding epiweek of date(D + 1).

Reviewer 3: So when someone requests this data, what is the `time_value`? I don't think we have any weekly signals yet, so it may be worth being explicit about how to interpret the `time_value` from the API.

Author: If the time_value in API is based on the file names, so they should be epiweeks. Not quite sure about this, need @krivard's answer.

Reviewer: `time_type=week&time_value={YYYYWW}`

[Canonical definition of CDC epiweeks](https://www.cmmcp.org/mosquito-surveillance-data/pages/epi-week-calendars-2008-2020#:~:text=WHAT%20IS%20AN%20%22EPI%20WEEK,of%20data%20year%20after%20year.) (I know it's a mosquito surveillance project but that seems to be the one everybody refers back to so 🤷 )

[Python library for converting between dates and epiweeks](https://pypi.org/project/epiweeks/); eg

```
from datetime import date
import epiweeks
print(epiweeks.Week.fromdate(date.today()))
```

## Data Versioning

> Data versions are tracked on both a daily and weekly level. On a daily level, we check for updates for NCHS mortality data every weekday as how it is reported by CDC and stash these daily updates on S3, but not our API. On a weekly level (on Mondays), we additionally upload the changes to the data made over the past week (due to backfill) to our API.

Reviewer 3: This is written from our perspective, not the user's. (For example, it doesn't matter to the user if we use S3 or not; it matters what revisions appear in the API and when.) We should be explicit: how does NCHS revise their data, and how do those revisions appear in our API?

## Lag and Backfill

> There is a lag in time between the death occurred and when the death certificate is completed, submitted to NCHS and processed for reporting purposes. This delay can range from 1 week to 8 weeks or even more, as described by NCHS. 

Reviewer 3: This is the first time death certificates are mentioned. If these reports are based on death certificates, perhaps we should state that right in the opening sentence, so users understand what kind of data this is?

Also, this paragraph should go on to say that when NCHS backfills data, we issue new numbers in our API, etc.
